packages/parameters/src/AppConfigProvider.ts

```diff
@@ -0,0 +1,111 @@
+import { BaseProvider, DEFAULT_PROVIDERS } from 'BaseProvider';
+import {
+  AppConfigDataClient,
+  StartConfigurationSessionCommand,
+  GetLatestConfigurationCommand,
+} from '@aws-sdk/client-appconfigdata';
+import type {
+  StartConfigurationSessionCommandInput,
+  GetLatestConfigurationCommandInput,
+} from '@aws-sdk/client-appconfigdata';
+import type { AppConfigGetOptionsInterface } from './types/AppConfigProvider';
+
+class AppConfigProvider extends BaseProvider {
+  public client: AppConfigDataClient;
+  private application?: string;
+  private environment?: string;
+  private token: string | undefined;
+
+  /**
+   * It initializes the AppConfigProvider class with an optional set of options like region: 'us-west-1'.
+   * *
+   * @param {AppConfigDataClientConfig} options
+   */
+  public constructor(options: AppConfigGetOptionsInterface = {}) {
+    super();
+    this.client = new AppConfigDataClient(options.config || {});
+    this.environment = options.environment;
+    this.application = options.application || 'service_undefined'; // TODO: get value from ENV VARIABLES
+  }
+
+  /**
+   * Retrieve a parameter value from AWS App config.
+   *
+   * @param {string} name - Name of the configuration
+   * @param {StartConfigurationSessionCommandInput} [sdkOptions] - SDK options to propagate to `StartConfigurationSession` API call
+   * @returns {Promise<string | undefined>}
+   */
+  protected async _get(
+    name: string,
+    options?: AppConfigGetOptionsInterface
+  ): Promise<string | undefined> {
+    /**
+     * The new AppConfig APIs require two API calls to return the configuration
+     * First we start the session and after that we retrieve the configuration
+     * We need to store the token to use in the next execution
+     **/
+    if (!this.token) {
+      const sessionOptions: StartConfigurationSessionCommandInput = {
+        ConfigurationProfileIdentifier: name,
+        EnvironmentIdentifier: this.environment,
+        ApplicationIdentifier: this.application,
+      };
+      if (options && options.hasOwnProperty('sdkOptions')) {
+        Object.assign(sessionOptions, options.sdkOptions);
+      }
+
+      const sessionCommand = new StartConfigurationSessionCommand(
+        sessionOptions
+      );
+
+      const session = await this.client.send(sessionCommand);
+      this.token = session.InitialConfigurationToken;
+    }
+
+    const getConfigurationCommand = new GetLatestConfigurationCommand({
+      ConfigurationToken: this.token,
+    });
+    const response = await this.client.send(getConfigurationCommand);
+
+    //  Should we cache and flush the token after 24 hours?
+    //  NextPollConfigurationToken expires in 24 hours after the last request and causes `BadRequestException`
```

Comment:
Is the token scoped to a specific configuration or can I retrieve multiple configurations with the same token?

I.e.: If I do `.get('configA')` and then `.get('configB')`, do they use the same token or do they need two different tokens?

Before making a decision on caching I'd like to understand this. By default we are already caching the response (in `BaseProvider`) but I can see how customers might want to hit AppConfig more times (i.e. retrieving multiple configs or refreshing the same config).

I'm not excessively concerned about the 24hrs duration given that we are optimizing for Lambda which is ephemeral in most cases. We expect most environments to never reach that lifespan but there could be edge cases in which an environment might last for longer spans of time (perhaps with provisioned concurrency). I would say that having a logic that refreshes the token in case of a `BadRequestException` would be a nice to have but not a must at the moment. I leave it up to you.

Reply:
This is a really good question. I opened an [issue](https://github.com/awslabs/aws-lambda-powertools-python/issues/1796) in the python repo about the function logic and going to adjust my implementation.

Reply:
Given that the Python's team is not expected to be 100% operational for [a little longer](https://github.com/awslabs/aws-lambda-powertools-python/issues/1784) I would move forward with the current implementation and revisit the implementation when the discussion picks up again.